teuthology/task/install/__init__.py

```diff
@@ -362,8 +365,10 @@ def upgrade_common(ctx, config, deploy_style):
         pkgs = get_package_list(ctx, config)[system_type]
         log.info("Upgrading {proj} {system_type} packages: {pkgs}".format(
             proj=project, system_type=system_type, pkgs=', '.join(pkgs)))
-        # FIXME: again, make extra_pkgs distro-agnostic
-        pkgs += extra_pkgs
+        if isinstance(extra_pkgs, dict):
+            pkgs += extra_pkgs.get(system_type)
```

Comment:
Why are we using ```system_type``` here but explicit ```'deb'``` or ```'rpm'``` strings above?

Reply:
Because here (in upgrade_common) we're looping over the remotes, and inside the loop we are already getting deb/rpm via `teuthology.get_system_type(remote)`. Whereas above, in the `install` function, we are not yet to the point of looping over remotes.

Reply:
Similar to Josh's comment, we should probably do `get(system_type, [])`